src/agent/executor/base.ts

```
@@ -0,0 +1,87 @@
+export abstract class OpenAIBaseExecutor extends BaseTaskExecutor {
+  static readonly log = createLogger('OpenAIBaseExecutor');
+  runtimeConfig: TaskRuntimeConfig = {} as TaskRuntimeConfig;
+  model: string | null = null;
+  openai: OpenAI | null = null;
+
+  async init() {
+    // This will be implemented by concrete classes
+    throw new Error('init() must be implemented by concrete executor');
+  }
+
+  async createOpenAIClient(configId: string) {
```

ESLint

[error] 'configId' is defined but never used.

(@typescript-eslint/no-unused-vars)